tests/unit/test1_unit_tests/src/main.c

```diff
@@ -95,6 +95,45 @@ void test_uint64(void)
 	zassert_true(zcbor_uint64_expect(&state_d, UINT64_MAX), NULL);
 }
 
+void test_size(void)
+{
+	uint8_t payload[100] = {0};
+	size_t read;
+
+	zcbor_state_t state_e;
+	zcbor_new_state(&state_e, 1, payload, sizeof(payload), 0);
+	zcbor_state_t state_d;
+	zcbor_new_state(&state_d, 1, payload, sizeof(payload), 10);
+
+	zassert_true(zcbor_size_put(&state_e, 5), NULL);
+	zassert_false(zcbor_size_expect(&state_d, 4), NULL);
+	zassert_false(zcbor_size_expect(&state_d, 6), NULL);
+	zassert_false(zcbor_size_expect(&state_d, -5), NULL);
+	zassert_false(zcbor_size_expect(&state_d, -6), NULL);
+	zassert_true(zcbor_size_expect(&state_d, 5), NULL);
+
+#if SIZE_MAX == UINT32_MAX
+	zassert_true(zcbor_uint32_put(&state_e, 5), NULL);
+	zassert_true(zcbor_size_expect(&state_d, 5), NULL);
+#endif
+
+#if SIZE_MAX == UINT64_MAX
+	zassert_true(zcbor_uint64_put(&state_e, 5), NULL);
+	zassert_true(zcbor_size_expect(&state_d, 5), NULL);
+#endif
+
+#if SIZE_MAX == UINT32_MAX
+	zassert_true(zcbor_uint32_put(&state_e, UINT32_MAX), NULL);
+	zassert_true(zcbor_size_decode(&state_d, &read), NULL);
+	zassert_equal(read, UINT32_MAX, NULL);
+#endif
```

Comment:
These three cases should all work for both SIZE_MAX values, so they don't need to be placed behind #ifs.

Reply:
Modified ifs.